### PR TITLE
add msccl++ support in README

### PR DESCRIPTION
**Pull request by nusislam**
## Details
___Do not mention proprietary info or link to internal work items in this PR.___

**Work item:** _"Internal", or link to GitHub issue (if applicable)._

**What were the changes?**  
_One sentence describing the work done._

**Why were the changes made?**  
_Explain the motivation behind the work. Provide any publicly-available historical context._

**How was the outcome achieved?**  
_Technical details behind the work. Explain any publicly-available hardware peculiarities._

**Additional Documentation:**  
_What else should the reviewer know?_

## Approval Checklist
___Do not approve until these items are satisfied.___
- [ ] Verify the CHANGELOG has been updated, if
  - there are any NCCL API version changes,
  - any changes impact library users, and/or
  - any changes impact any other ROCm library.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,6 +123,12 @@ To manually run RCCL with NPKit enabled, environment variable `NPKIT_DUMP_DIR` n
 
 To manually analyze NPKit dump results, please leverage [npkit_trace_generator.py](https://github.com/microsoft/NPKit/blob/main/rccl_samples/npkit_trace_generator.py).
 
+## MSCCL++
+RCCL integrates MSCCL++ (https://github.com/microsoft/mscclpp) to leverage the highly efficient GPU-GPU communication primitives for collective operations. Thanks to Microsoft Corporation for collaborating with us in this project.
+
+RCCL can be built with MSCCL++ support by providing the `-ENABLE_MSCCLPP` flag during compilation. RCCL allreduce and allgather collectives can leverage the efficient MSCCL++ communication kernels for certain message sizes. Users need to set the RCCL environment variable `RCCL_ENABLE_MSCCLPP=1` to run RCCL workload with MSCCL++ support.
```
Comment (corey-derochie-amd) [inline README.md]: The flag should be `-DENABLE_MSCCLPP=ON`. Also, should we document `RCCL_MSCCLPP_THRESHOLD`?

**Review by corey-derochie-amd — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (nileshnegi) [reply README.md]: maybe we should preface by saying use `--enable-mscclpp` when using the `./install.sh` script\
and, use `-DENABLE_MSCCLPP=ON` if using CMake directly

Comment (corey-derochie-amd) [reply README.md]: I don't think we have similarly complete docs around the other options available in `./install.sh`, `cmake`, or `make`. Perhaps we should, but for this, maybe we only just need to be consistent?

Comment (corey-derochie-amd) [reply README.md]: You're right, we should add it above.

Comment (nileshnegi) [reply README.md]: yeah, currently, only the `./install.sh --help` output is listed in README.md to show available options...\
we could extend this with more info in the README
P.S: no `make` for RCCL, only RCCL-Tests

**Review by wenkaidu — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -123,6 +123,14 @@ To manually run RCCL with NPKit enabled, environment variable `NPKIT_DUMP_DIR` n
...
+
+MSCCL uses XMLs for different collective algorithms on different architectures. RCCL collectives can leverage those algorithms once the corresponding XML has been provided by the user. The XML files contain the sequence of send-recv and reduction operations to be executed by the kernel. On MI300X, MSCCL is enabled by default. On other platforms, the users may have to enable this by setting `RCCL_MSCCL_FORCE_ENABLE=1`.
+
+On the other hand, RCCL can be built with MSCCL++ support by providing the `-DENABLE_MSCCLPP=ON` flag during compilation via `cmake`. When using the `install.sh` script, `--enable-mscclpp` flag should be used. RCCL allreduce and allgather collectives can leverage the efficient MSCCL++ communication kernels for certain message sizes. Users need to set the RCCL environment variable `RCCL_ENABLE_MSCCLPP=1` to run RCCL workload with MSCCL++ support. It is also possible to set the message size threshold for using MSCCL++ by using the environment variable `RCCL_MSCCLPP_THRESHOLD`. Once `RCCL_MSCCLPP_THRESHOLD` is set, RCCL will invoke MSCCL++ kernels for all message sizes less than or equal to the specified threshold.
```
Comment (corey-derochie-amd) [inline README.md]: Maybe consider documenting the default threshold value, but that might not be necessary.

**Review by corey-derochie-amd — APPROVED** [review]

**Comment by corey-derochie-amd** [comment]
Merged into https://github.com/ROCm/rccl/pull/1231 . Closing.

**Comment by nusislam** [comment]
@corey-derochie-amd - Why was this closed?

**Comment by corey-derochie-amd** [comment]
> @corey-derochie-amd - Why was this closed?

I've merged it into the other PR.

**Pull request closed without merging**